central/declarativeconfig/manager_impl.go

```diff
@@ -236,7 +238,9 @@ func (m *managerImpl) runReconciliation() {
 }
 
 func (m *managerImpl) reconcileTransformedMessages(transformedMessagesByHandler map[string]protoMessagesByType) {
-	log.Debugf("Run reconciliation for the next handlers: %v", maps.Keys(transformedMessagesByHandler))
+	if logging.GetGlobalLogLevel().Enabled(logging.DebugLevel) {
+		log.Debugf("Run reconciliation for the next handlers: %v", slices.Collect(maps.Keys(transformedMessagesByHandler)))
+	}
```

Comment:
There's a little problem here: `log` does not necessarily use the global level.

An alternative approach could be something around this:
```go
	if w := log.SugaredLogger().Desugar().Check(zapcore.DebugLevel, "debugging"); w != nil {
		w.Write(
			zap.String("message", "Run reconciliation for the next handlers"),
			zap.Strings("handlers", maps.Keys(transformedMessagesByHandler)),
		)
	}
```

Maybe we could have a `IsDebugLevel(log)` helper for such condition.

Reply:
I think we can also remove this line 😉

Reply:
```suggestion
		log.Debugf("Run reconciliation for the next handlers: %d", len(transformedMessagesByHandler))
```